packages/admin/cms-admin/src/pages/pagesPage/PagesPage.tsx

```diff
@@ -39,7 +39,13 @@ interface Props {
     category: string;
     path: string;
     allCategories: AllCategories;
-    documentTypes: Record<DocumentType, DocumentInterface>;
+    /**
+     * @deprecated Use categoryToDocumentTypesMap instead
+     */
+    documentTypes?: Record<DocumentType, DocumentInterface>;
+    categoryToDocumentTypesMap?: {
+        [category: string]: Record<DocumentType, DocumentInterface>;
+    };
```

Comment:
Wouldn't it be nicer to still have documentTypes, but support both cases? There are many projects that don't have different document types per category. Maybe something like: `documentTypes?: Record<DocumentType, DocumentInterface> | Record<string, Record<DocumentType, DocumentInterface>>` or `documentTypes?: Record<DocumentType, DocumentInterface> | (category: string) => Record<DocumentType, DocumentInterface>`?

Reply:
Sounds good - https://github.com/vivid-planet/comet/pull/2116/commits/b82fae1793f23327120fd0b2ef8e3cb79707af36